xtraplatform-feature-provider-sql/src/main/java/de/ii/xtraplatform/feature/provider/sql/infra/db/SqlTypeInfoValidator.java

```diff
@@ -61,25 +63,29 @@ public SqlTypeInfoValidator(ConnectionInfoSql connectionInfo, ClassLoader classL
             errors.add(String.format("table '%s' does not exist", attributesContainer.getName()));
         }
 
-        if (!columnExists( attributesContainer.getSortKey(), attributesContainer.getName())) {
+        if (!columnExists(attributesContainer.getSortKey(), attributesContainer.getName())) {
             errors.add(String.format("column '%s' in table '%s' does not exist", attributesContainer.getSortKey(), attributesContainer.getName()));
         }
 
-        //TODO: isColumnUnique( attributesContainer.getSortKey(), attributesContainer.getName())
-        // -> check that column is either the primary key of the table or that a unique index for the column exists
+        if (!isColumnUnique(attributesContainer.getSortKey(), attributesContainer.getName())) {
+            errors.add(String.format("column %s in table %s is not a primary key or a unique index for the column does not exist",
+                    attributesContainer.getSortKey(), attributesContainer.getName()));
+        }
+
 
         attributesContainer.getAttributes().forEach(attribute -> {
             if (!attribute.isConstant()) {
-                //TODO: columnExists( attribute.getName(), attributesContainer.getName())
+                if (!columnExists(attribute.getName(), attributesContainer.getName())) {
+                    errors.add(String.format("column '%s' in table '%s' does not exist", attribute.getName(), attributesContainer.getName()));
+                }
 
-                if (attribute.isId()) {
-                    //TODO: isColumnUnique( attribute.getName(), attributesContainer.getName())
-                    // -> check that column is either the primary key of the table or that a unique index for the column exists
+                if (attribute.isId() && !isColumnUnique(attribute.getName(), attributesContainer.getName())) {
+                    errors.add(String.format("column %s in table %s is not a primary key or a unique index for the column does not exist",
+                            attributesContainer.getSortKey(), attributesContainer.getName()));
                 }
 
-                if (attribute.isSpatial()) {
-                    //TODO: isColumnSpatial( attribute.getName(), attributesContainer.getName())
-                    // -> check that column is of type geometry
+                if (attribute.isSpatial() && !isColumnSpatial(attribute.getName(), attributesContainer.getName())) {
+                    errors.add(String.format("column %s in table %s is not of type geometry", attribute.getName(), attributesContainer.getName()));
                 }
```

Comment:
Add similar check with `attribute.isTemporal()` (maybe you have to pull, I added this later) and `isColumnTemporal` (to implement, see interactive-instruments/ldproxy#260).